src/Entity/DispatchingEntityIdParser.php

```diff
@@ -37,20 +41,19 @@ public function __construct( array $idBuilders ) {
 	 * @return EntityId
 	 */
 	public function parse( $idSerialization ) {
-		$this->assertIdIsString( $idSerialization );
+		try {
+			list( , , $localId ) = EntityId::splitSerialization( $idSerialization );
+		} catch ( InvalidArgumentException $ex ) {
+			// Only do this assert in case something is wrong, EntityId::splitSerialization already
+			// checks for string.
+			$this->assertIdIsString( $idSerialization );
```

Comment:
Why keep this additional check *inside* the catch? This looks awkward. What does this do in addition to what splitSerialization already does? Let's remove it, please.

Reply:
changing it will change the Exception message slightly, but I guess that's ok.